packages/unlock-js/src/PublicLock/utils/multiplePurchaseWrapper.js

```diff
@@ -40,19 +40,25 @@ export default async function (
     erc20Address = await lockContract.tokenAddress()
     // if ERC20 is set we approve the entire amount
     if (erc20Address !== ZERO) {
+      // We might not have the keyPrice, in which case, we need to retrieve from the the lock!
+      const getPrice = async (price) =>
+        !price
+          ? await lockContract.keyPrice()
+          : await formatKeyPrice(price, erc20Address, decimals, this.provider)
+
       const prices = await Promise.all(
-        (keyPrices || Array(owners.length).fill(null)).map(async (kp) =>
```

Comment:
for some reasons, the previous syntax seems to lead to failure when passing no explicit prices with ERC20 locks